File: solutions/fsharp/kindergarten-garden/1/KindergartenGarden.fs
module KindergartenGarden

type Plant =
    | Radishes
    | Clover
    | Violets
    | Grass

let students =
    [
        "Alice"; "Bob"; "Charlie"; "David"
        "Eve"; "Fred"; "Ginny"; "Harriet"
        "Ileana"; "Joseph"; "Kincaid"; "Larry" ]
    |> List.mapi (fun idx stu -> (idx + 1, stu)) // account for zero based index

let translate c =
    match c with
    | 'R' -> Radishes
    | 'V' -> Violets
    | 'G' -> Grass
    | 'C' -> Clover
    | _ -> failwith "No no"

let plants (diagram: string) (student: string) =

    let indices =
        students
        |> List.filter (fun stu -> snd stu = student)
        |> List.head
        |> fun (idx, _) ->
            let double = idx * 2
            [ double - 1 .. double ]

    diagram.Split('\n')
    |> Seq.map (Seq.map translate)
    |> Seq.collect (Seq.mapi (fun idx plant -> (idx + 1, plant)))
    |> Seq.filter (fun xs -> indices |> List.contains (fst xs))
    |> Seq.map snd
    |> List.ofSeq

File: solutions/csharp/kindergarten-garden/1/KindergartenGarden.cs
public enum Plant { Violets = 1, Radishes, Clover, Grass }

public class KindergartenGarden(string diagram)
{
    static Plant GetPlant(char c) => c switch
    {
        'R' => Plant.Radishes,
        'V' => Plant.Violets,
        'G' => Plant.Grass,
        'C' => Plant.Clover,
        _ => throw new Exception()
    };

    readonly List<string> _students =
        [ "Alice", "Bob", "Charlie", "David",
          "Eve", "Fred", "Ginny", "Harriet",
          "Ileana", "Joseph", "Kincaid", "Larry" ];

    readonly string[] _diagram = diagram.Split('\n');

    public Plant[] Plants(string student) =>
        [.. _diagram
            .SelectMany(x => x
                .Skip(_students.IndexOf(student) * 2)
                .Take(2)
                .Select(GetPlant))];
}